powersimdata/input/change_table.py

```diff
@@ -789,17 +794,92 @@ def _get_df_with_new_elements(self, table):
             'branch', 'bus', 'dcline', or 'plant'.
         :return: (*pandas.DataFrame*) -- the post-transformation table.
         """
-        add_key = f"new_{table}"
-        if add_key not in self.ct:
+        modification_keys = [f"new_{table}"]
+        if table == "plant":
+            modification_keys += sorted(self.grid.plant["type"].unique())
+        if not any(m in self.ct for m in modification_keys):
             return getattr(self.grid, table)
-        new_elements_tuple = tuple(tuple(sorted(b.items())) for b in self.ct[add_key])
-        if new_elements_tuple in self._new_element_caches[table]:
-            return self._new_element_caches[table][new_elements_tuple]
+        modified_elements_tuple = tuple(
+            chain.from_iterable(
+                [
+                    f"new_{table}",
+                    tuple(tuple(sorted(b.items())) for b in self.ct.get(table, {})),
+                ]
+                + [
+                    tuple([subkey] + sorted(subdict.items()))
+                    for k in modification_keys[1:]
+                    for subkey, subdict in self.ct.get(k, {}).items()
+                ]
+            )
+        )
```

Comment:
you lost me here. What if `modification` is a single element, i.e., `table` is not `plant`? Won't `modification_keys[1:]` raise an error?

Reply:
No, we'll just get an empty list:
```python
>>> foo = ["a"]
>>> foo[1:]
[]
```

The current implementation is clunky for sure. If `table == "plant"`, then we want to reliably create a tuple that includes all plant additions and plant scaling information. If `table != "plant"`, we should be creating a tuple as before, just with one new top-level string entry for `"new_bus"` or `"new_branch"` or whatever.